### DISCUSSION TITLE
Is it possible to have 1 wildcards folder for many A1111 installs?

### DISCUSSION THREAD

**Discussion by ArnorWing** [adieyal/sd-dynamic-prompts#812] [Q&A] [answered]
I have many installs of A1111 and Forge. 
Is it possible just to have 1 central folder with all wildcards instead of continually copying the files to each installs extension folder?

**Comment by SvenErik1968** [comment]
Yes, you can use [symlinks](https://ss64.com/nt/mklink.html); creating a symbolic link that points to a folder (or file).

Example (_Windows_), you have A1111 in the folder `D:\SD\stable-diffusion-webui` and Forge in the folder `D:\SD\stable-diffusion-webui-forge`. In both cases, the wildcards folder will be in `extensions\sd-dynamic-prompts\wildcards`. And you want your shared wildcard folder in `D:\SD\Resources\wildcards`.

Move any content in the wildcard folders (`D:\SD\stable-diffusion-webui\extensions\sd-dynamic-prompts\wildcards` and `D:\SD\stable-diffusion-webui-forge\extensions\sd-dynamic-prompts\wildcards`) to the shared wildcard folder, then delete the now empty wildcard folders. Open a command window **with administrator rights**, then type the following:
```
mklink /J "D:\SD\stable-diffusion-webui\extensions\sd-dynamic-prompts\wildcards" "D:\SD\Resources\wildcards"
mklink /J "D:\SD\stable-diffusion-webui-forge\extensions\sd-dynamic-prompts\wildcards" "D:\SD\Resources\wildcards"
```
Now you will again have a wildcards folder under each installation of sd-dynamic-prompts, think of them as _separate windows_ into the `D:\SD\Resources\wildcards` shared folder, so any changes you do in the `D:\SD\stable-diffusion-webui\extensions\sd-dynamic-prompts\wildcards` will be the same as if you did it in the `D:\SD\Resources\wildcards` folder.

You can use the same method for your `models` and `embeddings` folders allowing you to save a lot of storage space by not duplicating files all the time, and you don't have to mess around with defining paths in the webui-user.bat file.

**Reply by ArnorWing** [reply]
Thank you for your in depth answer.
I do know about symlinks.
 I was hoping for a solution like controlnet has, in A1111/forge settings the path for other folders can be entered. 
I use the .bat file edit method with all my installs for /models/ folder which is 2TB now haha. I just copy paste paths or even just C&P the whole .bat file. 
With wildcards folder i just C&P it. its only 9MB.

**Comment by valtnsfw** [comment, marked as answer]
In the config.json for your webui install add this line:
"wildcard_dir": "C:\\Users\\mrval\\AppData\\Roaming\\StabilityMatrix\\wildcards",
(replacing your file path for your wildcard folder for mine) Dynamic Prompts automatically recognizes it.
[Full info near the bottom of the README](https://github.com/adieyal/sd-dynamic-prompts?tab=readme-ov-file#wildcard_dir)

**Reply by ArnorWing** [reply]
Thanks for that. I would've looked in the readme but missed this answer.
Thanks again
